api/src/main/scala/za/co/absa/loginsvc/rest/config/auth/ActiveDirectoryLDAPConfig.scala

```diff
@@ -57,3 +72,110 @@ case class ActiveDirectoryLDAPConfig(domain: String, url: String, searchFilter:
     else ConfigValidationSuccess
   }
 }
+
+case class ServiceAccountConfig(private val accountPattern: String,
+                                private val inConfigAccount: Option[IntegratedLdapUserConfig],
+                                private val awsSecretsManagerAccount: Option[AwsSecretsLdapUserConfig])
+{
+  private val ldapUserDetails: LdapUser = (inConfigAccount, awsSecretsManagerAccount) match {
+    case (Some(_), Some(_)) =>
+      throw ConfigValidationException("Both integratedLdapUserConfig and awsSecretsLdapUserConfig exist. Please choose only one.")
+
+    case (None, None) =>
+      throw ConfigValidationException("Neither integratedLdapUserConfig nor awsSecretsLdapUserConfig exists. One of them should exist.")
+
+    case _ =>
+      val ldapConfig: LdapUser = inConfigAccount.orElse(awsSecretsManagerAccount).get
+      ldapConfig.throwErrors()
+
+      ldapConfig
+  }
+
+  val username: String = String.format(accountPattern, ldapUserDetails.username)
+  val password: String = ldapUserDetails.password
+}
+
+case class IntegratedLdapUserConfig(username: String, password: String) extends LdapUser
```

Comment:
How about just `LdapUserCredetialsConfig` or similar - the pair is essentially just _credentials_.

Reply:
The original name `integratedLdapUserConfig` has been left in logging string (`ActiveDirectoryLDAPConfig.scala`).

Reply:
Will fix this now

Reply:
Almost, I see still one left in `throw ConfigValidationException("Neither integratedLdapUserConfig nor awsSecretsLdapUserConfig exists. Exactly one of them should be present.")`